src/lib/dateHandler.ts

```diff
@@ -0,0 +1,100 @@
+/**
+ * The TranslateMonth type
+ */
+type TranslateMonth = (month: number) => string;
+
+/**
+ * DateHandler class
+ */
+export class DateHandler {
+  /**
+   * Get formatted date as string with default date format "dd.MM.yyyy"
+   * @param dateString The date string to format
+   * @returns Date string formatted
+   */
+  public static getDateString(dateString: string): string {
+    return this.getDateFormatted(dateString, "dd.MM.yyyy");
+  }
+
+  /**
+   * Get formatted date as string with custom date format
+   * @param dateString The date string to format
+   * @param dateFormat The date format
+   * @param defaultString The default string
+   * @returns Date string formatted
+   */
+  public static getDateFormattedWithDefault(dateString: string | Date, dateFormat: string, defaultString: string): string {
+    if (dateString === null || dateString === undefined) {
+      return defaultString;
+    }
+    return this.getDateFormatted(dateString, dateFormat);
+  }
+
+  /**
+   * Format date string
+   * @param dateString The date string to format
+   * @param dateFormat The date format
+   * @returns Date string formatted
+   */
+  public static getDateFormatted(dateString: string | Date, dateFormat: string): string {
+    const date = new Date(dateString);
+    let formattedDateString = dateFormat;
+
+    if (dateFormat.indexOf("MMMM") >= 0) {
+      formattedDateString = formattedDateString.replace("MMMM", this.translateMonthMethod(date.getMonth() + 1));
+    }
+
+    return formattedDateString
+      .replace("yyyy", date.getFullYear().toString())
```

Comment:
I don't think it makes sense to have code like this. 

I'm sure that there is a well maintained and tested library that takes care of this with all the funny edge cases... 

What's your opinon @neoscie and @neotob